diff -Nur -x '*~' -x '*.orig' -x '*.rej' fposs-290518/src/Makefile fposs/src/Makefile
--- fposs-290518/src/Makefile	2016-11-24 16:23:24.000000000 +0100
+++ fposs/src/Makefile	2016-12-13 10:03:18.000000000 +0100
@@ -108,24 +108,30 @@
 
 install :
 	@# for simplicity, we assume that INTROOT is always available
-	@/bin/rm -f $(INTROOT)/bin/configure
-	@/bin/cp ../bin/configure $(INTROOT)/bin/configure
-	@/bin/rm -f $(INTROOT)/bin/configure_expert
-	@/bin/cp ../bin/configure_expert $(INTROOT)/bin/configure_expert
-	@/bin/rm -f $(INTROOT)/lib/configure
-	@/bin/cp ../lib/configure $(INTROOT)/lib/configure
-	@/bin/rm -f $(INTROOT)/lib/configure_expert
-	@/bin/ln -s configure $(INTROOT)/lib/configure_expert
-	@/bin/rm -f $(INTROOT)/lib/configure.tcl
-	@/bin/cp ../lib/configure.tcl $(INTROOT)/lib/configure.tcl
-	@/bin/rm -f $(INTROOT)/lib/mimic.tcl
-	@/bin/cp ../lib/mimic.tcl $(INTROOT)/lib/mimic.tcl
-	@/bin/rm -f $(INTROOT)/lib/procs.tcl
-	@/bin/cp ../lib/procs.tcl $(INTROOT)/lib/procs.tcl
-	@/bin/rm -f $(INTROOT)/lib/util.tcl
-	@/bin/cp ../lib/util.tcl $(INTROOT)/lib/util.tcl
-	@/bin/rm -f $(INTROOT)/config/fpossInsStatus.sds
-	@/bin/cp ../config/fpossInsStatus.sds $(INTROOT)/config/fpossInsStatus.sds
+	@/usr/bin/install -d "$(INTROOT)"/lib/fposs/bin
+	@/usr/bin/install -d "$(INTROOT)"/lib/fposs/lib
+	@/usr/bin/install -d "$(INTROOT)"/lib/fposs/config
+	@/usr/bin/install -d "$(INTROOT)"/share/fposs
+	@/bin/rm -f "$(INTROOT)"/lib/fposs/bin/configure
+	@/usr/bin/install ../bin/configure "$(INTROOT)"/lib/fposs/bin/configure
+	@/bin/rm -f "$(INTROOT)"/lib/fposs/bin/configure_expert
+	@/usr/bin/install ../bin/configure_expert "$(INTROOT)"/lib/fposs/bin/configure_expert
+	@/bin/rm -f "$(INTROOT)"/lib/fposs/lib/configure
+	@/usr/bin/install ../lib/configure "$(INTROOT)"/lib/fposs/lib/configure
+	@/bin/rm -f "$(INTROOT)"/lib/fposs/lib/configure_expert
+	@/bin/ln -s configure "$(INTROOT)"/lib/fposs/lib/configure_expert
+	@/bin/rm -f "$(INTROOT)"/lib/fposs/lib/configure.tcl
+	@/usr/bin/install ../lib/configure.tcl "$(INTROOT)"/lib/fposs/lib/configure.tcl
+	@/bin/rm -f "$(INTROOT)"/lib/fposs/lib/mimic.tcl
+	@/usr/bin/install ../lib/mimic.tcl "$(INTROOT)"/lib/fposs/lib/mimic.tcl
+	@/bin/rm -f "$(INTROOT)"/lib/fposs/lib/procs.tcl
+	@/usr/bin/install ../lib/procs.tcl "$(INTROOT)"/lib/fposs/lib/procs.tcl
+	@/bin/rm -f "$(INTROOT)"/lib/fposs/lib/util.tcl
+	@/usr/bin/install ../lib/util.tcl "$(INTROOT)"/lib/fposs/lib/util.tcl
+	@/bin/rm -f "$(INTROOT)"/lib/fposs/config/fpossInsStatus.sds
+	@/usr/bin/install ../config/fpossInsStatus.sds "$(INTROOT)"/lib/fposs/config/fpossInsStatus.sds
+	@/bin/rm -fr "$(INTROOT)"/share/fposs/test
+	@/usr/bin/rsync -ax ../test "$(INTROOT)"/share/fposs
 	@echo " . . . installation done."
 
 #*******************************************************************************
diff -Nur -x '*~' -x '*.orig' -x '*.rej' fposs-290518/src/configure.tcl fposs/src/configure.tcl
--- fposs-290518/src/configure.tcl	2016-11-24 16:23:24.000000000 +0100
+++ fposs/src/configure.tcl	2016-12-13 10:03:18.000000000 +0100
@@ -8854,8 +8854,8 @@
     set config_font_1 {Times 14 }
     set config_font_2 {Times 12 }
 } else {
-    set config_font_1 "-Adobe-Times-Medium-R-Normal--*-180-*"
-    set config_font_2 "-Adobe-Helvetica-Medium-R-Normal--*-120-*"
+    set config_font_1 {Times 14 }
+    set config_font_2 {Times 12 }
 }
 
 #  Create the main display window, menus and mimic display etc.
diff -Nur -x '*~' -x '*.orig' -x '*.rej' fposs-290518/src/mimic.tcl fposs/src/mimic.tcl
--- fposs-290518/src/mimic.tcl	2015-02-07 23:15:22.000000000 +0100
+++ fposs/src/mimic.tcl	2016-12-13 10:03:18.000000000 +0100
@@ -198,8 +198,8 @@
     set MimicShowFont {Helvetica 12 }
     set SmallShowFont {Helvetica 9 }
 } else {
-    set MimicShowFont "-Adobe-Helvetica-Medium-R-Normal--*-120-*"
-    set SmallShowFont "-Adobe-Helvetica-Medium-R-Normal--*-120-*"
+    set MimicShowFont {Helvetica 12 }
+    set SmallShowFont {Helvetica 9 }
 }
 
 #  Set the globals connected with guider selection. These all are set to the
diff -Nur -x '*~' -x '*.orig' -x '*.rej' fposs-290518/src/procs.tcl fposs/src/procs.tcl
--- fposs-290518/src/procs.tcl	2015-02-07 23:15:22.000000000 +0100
+++ fposs/src/procs.tcl	2016-12-13 10:03:18.000000000 +0100
@@ -40,7 +40,7 @@
            ($tcl_platform(platform) == "macintosh")} {
     set procs_font_1 {Times 14 }
 } else {
-    set procs_font_1 "-Adobe-Times-Medium-R-Normal--*-180-*"
+    set procs_font_1 {Times 14 }
 }
  
 #
